pkg/bpf/bpffs_migrate.go

```diff
@@ -185,10 +187,50 @@ func finalizeMap(bpffsPath, name string, revert bool) error {
 		return nil
 	}
 
+	// Never unpin prog arrays. Doing so causes existing BPF progs to miss tail
+	// calls, resulting in packet drops on some interfaces. Instead, move the pin
+	// to a backup directory to prevent the array from being deallocated.
+	// There are only few, they're limited in size and only need to be recreated
+	// when an agent upgrade changes the array size, so don't garbage collect
+	// them. This causes less overhead than regularly scanning all progs for
+	// references to determine if it's safe to delete.
+	if pending.Type() == ebpf.ProgramArray {
+		log.WithFields(logrus.Fields{logfields.BPFMapPath: file, logfields.BPFMapName: name}).
+			Info("Backing up program array after successful recreation")
```

Comment:
Do we really need to log this at info level? Seems like a very specific implementation detail which means nothing to users.

Reply:
This feels like a crucial event to have in the logs as it's only performed during upgrades when map properties have changed. Especially given the troubleshooting history that led to this PR, being able to observe this event (or a lack thereof) could be important in some situations.

I'll leave this up to @brb or @borkmann to decide, but will leave it in for now.

Reply:
I'd keep the log statement at least until we have enough confidence that the new map migrate infrastructure is sound.